src/App.js

```diff
@@ -1,28 +1,42 @@
-import logo from './logo.svg';
-import './App.css';
+import { useDispatch } from 'react-redux';
+import { useEffect } from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Routes, Route, Navigate } from 'react-router';
+import SignUp from './pages/Sign_up';
+import SignIn from './pages/Sign_in';
+import GameDetails from './pages/GameDetailsPage';
+import dispatchGetGames from './store/slices/get_games_slice';
+import ReservationPage from './pages/ReservationPage';
+import Layout from './Layout';
+import HomePage from './pages/HomePage';
+import MyReservations from './pages/MyReservations';
+import AddGame from './pages/AddGame';
+import GamesList from './pages/GamesList';
 
 function App() {
+  const dispatch = useDispatch();
+  const user = localStorage.getItem('userInfo');
+  useEffect(() => {
+    dispatchGetGames(dispatch);
+  }, []);
```

Comment:
- [ ] **(OPTIONAL)** I wouldn't use this logic here in the `App` component, I believe it would be better if you check this in a helper function outside of this component. Also, you could use an attribute called `onEnter` which you can put in each `<Route>` you need authentication.

Reply:
Hey Murilo! I searched this onEnter hook, apparently, it does not exist anymore on react-router v6. And when you talk about 'this logic' do you refer to only the dispatchGetGames? I am thinking of putting it inside a helper but would it not be the same? Calling `dispatchGetGames` and `getGames` function for example.

Reply:
Hey @jmontanholi, I didn't know that!
The helper would have been helpful if we could use the hook, but in this case I think it's fine like this 🙃

Reply:
Oh ok, thanks for the review by the way! Have a nice day!